# Case 1: Straight duct with unchanged velocity

### Goal

Demonstrate the function can calculate the $$\sum_{\text{out}}\dot{m}\mathbf{V} - \sum_{\text{in}}\dot{m}\mathbf{V}$$ correctly.

### Assumptions and given conditions

1. The flow is steady.
2. The fluid is incompressible.
3. The inlet's mass flow rate and the outlet's mass flow rate are the same.
4.  $\dot{m}=4.0\ \mathrm{kg/s}$, $\mathbf{V}_{\text{in}}=(3.0,0,0)\ \mathrm{m/s}$, and $\mathbf{V}_{\text{out}}=(3.0,0,0)\ \mathrm{m/s}$.

### We have to get,

1. net momentum flux vector, $\mathbf{P}_\text{net}$
2. Why does the $\mathbf{P}_\text{net}$ become zero?
3. Why can pressure force and frictional force exist when $\mathbf{P}_\text{net} = 0$

In [1]:
from src.momentum_tool import(
  mass_flow_rate,
    net_momentum_flux,
)
import numpy as np

In [2]:
# case 1
mdot = 4.0

inlets = [{"velocity": (3.0, 0, 0), "mass_flow_rate": mdot}]
outlets = [{"velocity": (3.0, 0, 0), "mass_flow_rate": mdot}] # [m/s]

p_net = net_momentum_flux(inlets, outlets)

print(f" P_net = {p_net}")

 P_net = [0. 0. 0.]


### Results

$$\mathbf{P}_\text{net} = [0. 0. 0.]\, \mathrm{N}$$

### Physical Interpretation

비압축성 유체가 일정한 단면을 가진 검사체적에서 직선 정상유동을 할 때, 검사체적에 출입하는 유체의 단면 평균 속도는 같고,  $$\int_{CS} \rho \mathbf{V}(\mathbf{V} \cdot \mathbf{n}) dA$$
으로 나타내는 순 운동량 플럭스가 $0$이 된다.

When an incompressible fluid undergoes steady, straight flow through a control volume with a constant cross-sectional area, the cross-sectional average velocities of the fluid entering and leaving the control volume are the same. Therefore, the net momentum flux expressed as

$$
\int_{\mathrm{CS}}
\rho\mathbf{V}(\mathbf{V}\cdot\mathbf{n})\,dA
$$

becomes zero.

### Answer the questions
Q1. Why does the $\mathbf{P}_\text{net}$ become zero?

A1. 정상 유동 아래 검사체적의 질량은 보존되기에 유입되는 질량과 유출되는 질량이 같고, 일정한 단면적을 가지기에 입구와 출구의 유속 또한 동일하다. 따라서 $\sum_{\text{out}}\dot{m}\mathbf{V} - \sum_{\text{in}}\dot{m}\mathbf{V}$으로 계산되는 순 운동량 플럭스는 0이 된다.

Under steady flow, mass is conserved within the control volume, so the mass entering and leaving the control volume is the same. Since the inlet and outlet also have the same cross-sectional area, their velocities are the same. Therefore, the net momentum flux calculated by

$$
\sum_{\text{out}}\dot{m}\mathbf{V}
-
\sum_{\text{in}}\dot{m}\mathbf{V}
$$

becomes zero.

------------------------------------------------------

Q2. Why can pressure force and frictional force exist when $\mathbf{P}_\text{net} = 0$

A2. 레이놀즈 수송정리와 뉴턴의 제 2법칙을 활용하여 선형 운동량 방정식을 도출할 때,

   $$\sum \vec{F}_\text{c.v} = \frac{\partial}{\partial t} \int_{\text{CV}} \vec{V} \rho \, dV + \int_{\text{CS}} \vec{V} \rho(\mathbf{V} \cdot \mathbf{n}) \, dA $$

을 얻게되고, 정상 유동의 경우 

$$\sum \vec{F}_\text{c.v} = \int_{\text{CS}} \vec{V} \rho(\mathbf{V} \cdot \mathbf{n}) \, dA $$
을 얻는다. 이때 우변이 0이 되더라도 좌변은 알짜힘이기에, 압력력과 마찰력의 벡터합이 0이 된다면 충분히 압력력과 마찰력이 존재할 수 있다.

When the linear momentum equation is derived using the Reynolds transport theorem and Newton's second law, we obtain

$$
\sum\mathbf{F}_{\mathrm{CV}}
=
\frac{\partial}{\partial t}
\int_{\mathrm{CV}}\rho\mathbf{V}\,dV
+
\int_{\mathrm{CS}}
\rho\mathbf{V}(\mathbf{V}\cdot\mathbf{n})\,dA.
$$

For steady flow, this becomes

$$
\sum\mathbf{F}_{\mathrm{CV}}
=
\int_{\mathrm{CS}}
\rho\mathbf{V}(\mathbf{V}\cdot\mathbf{n})\,dA.
$$

Even if the right-hand side is zero, the left-hand side represents the net force. Therefore, pressure and friction forces can still exist as long as their vector sum is zero.


# Case 2: Accelerating nozzle flow

### Goal

Check the relation between speed and momentum flux.

### Assumptions and given conditions

1. The flow is steady.
2. The fluid is incompressible.
3. The nozzle is horizontal
4. The fluid flows in the positive x-direction.
5. $\rho = 1000 kg/m^3 $
6. inlet : $A_1 = 0.010 m^2, \mathbf{V}_1 = 2.0 m/s$
7. outlet: : $A_2 = 0.004 m^2 $
   

### We have to get,

1. mass flow rate, $\dot{m}$
2. outlet velocity $\mathbf{V}_2$
3. inlet momentum flux vector, $\mathbf{P}_\text{in}$
4. outlet momentum flux vector, $\mathbf{P}_\text{out}$
5. net momentum flux vector, $\mathbf{P}_\text{net}$

### Question

1. Why is the outlet speed bigger than the inlet speed?
2. Is the x-component of the net momentum flux positive?
3. What direction of external force does this result indicate?
4. Why do we have to consider pressure force to obtain the actual force?

In [3]:
# case 2

rho = 1000 # [kg/m^3]

A_1 = 0.010 # [m^2]
V_1 = 2.0 # [m/s]

A_2 = 0.004 # [m^2]

V_2 = A_1*V_1/A_2

mdot = mass_flow_rate(rho, A_1, V_1)

inlets = [ {"velocity": (V_1, 0, 0), "mass_flow_rate": mdot} ]
outlets = [ {"velocity": (V_2, 0, 0), "mass_flow_rate": mdot} ]

net = net_momentum_flux(inlets, outlets)

print(f"mass flow rate: {mdot}, V_2: {V_2}, inlet_momentum_flux: {V_1*mdot}, outlet_momentum_flux: {V_2*mdot}, net_momentum_flux: {net}")

mass flow rate: 20.0, V_2: 5.0, inlet_momentum_flux: 40.0, outlet_momentum_flux: 100.0, net_momentum_flux: [60.  0.  0.]


### Results

$$
\dot{m}=20.0\ \mathrm{kg/s},
\qquad
\mathbf{V}_2=(5.0,0,0)\ \mathrm{m/s}
$$

$$
\mathbf{P}_{\text{in}}=(40.0,0,0)\ \mathrm{N},
\qquad
\mathbf{P}_{\text{out}}=(100.0,0,0)\ \mathrm{N}
$$

$$
\mathbf{P}_{\text{net}}=(60.0,0,0)\ \mathrm{N}
$$

### Physical Interpretation

속도의 크기가 변하면 $$\int_{\text{CS}} \vec{V} \rho(\mathbf{V} \cdot \mathbf{n}) \, dA $$로 정의되는 운동량 플럭스에 변화가 생긴다. 정상 유동 아래 질량 유량은 유입량과 유출량이 동일하기에, 운동량 플럭스의 크기는 출입구의 속도에 비례한다. 출구의 단면적이 입구의 단면적 보다 작다면, 출구의 유속은 입구의 유속보다 크고, 이는 출구 운동량 플럭스가 입구 운동량 플럭스 보다 큼을 의미하며, x축 양의 방향으로 움직이는 case 2의 경우, 순운동량 플럭스가 양의 값을 가진다.

When the magnitude of velocity changes, the momentum flux defined by

$$
\int_{\mathrm{CS}}
\rho\mathbf{V}(\mathbf{V}\cdot\mathbf{n})\,dA
$$

also changes. Under steady flow, the inlet and outlet mass flow rates are the same, so the magnitude of the momentum flux is proportional to the velocity at each opening. If the outlet area is smaller than the inlet area, the outlet velocity is greater than the inlet velocity. This means that the outlet momentum flux is greater than the inlet momentum flux. Since the flow in Case 2 moves in the positive $x$-direction, the net momentum flux has a positive $x$-component.


### Answer the questions
Q1. Why is the outlet speed bigger than the inlet speed?

A1. 비압축성 정상 유동하에서 검사체적의 질량은 보존되므로, 질량 유입률과 질량 유출률은 동일해야 한다. 따라서 $A_1V_1=A_2V_2$을 만족시켜야 하며 이로부터 $V_2=A_1V_1/A_2$를 얻는다. 

$\frac{A_1}{A_2}>1$이기에, $V_2>V_1$의 결과를 얻는다.

Under steady incompressible flow, mass is conserved within the control volume, so the inlet and outlet mass flow rates must be the same. Therefore,

$$
A_1V_1=A_2V_2
$$

must be satisfied, and this gives

$$
V_2=\frac{A_1V_1}{A_2}.
$$

Since $\frac{A_1}{A_2}>1$, we obtain $V_2>V_1$.

--------------------------------------

Q2. Is the x-component of the net momentum flux positive?

A2. 그렇다. $\mathbf{P}_\text{net}: [60.  0.  0.]$에서 확인할 수 있다.

Yes. This can be confirmed from

$$
\mathbf{P}_{\text{net}}=(60.0,0,0)\ \mathrm{N}.
$$

--------------------------------------

Q3. What is the external force's direction that this result means?

A3. $$\mathbf{P}_\text{net}: [60.  0.  0.]$$이기에, 

$$\sum \vec{F}_\text{c.v} = \int_{\text{CS}} \vec{V} \rho(\mathbf{V} \cdot \mathbf{n}) \, dA $$
에 따라 $x+$ 방향의 외력이 필요함을 의미한다.

Since

$$
\mathbf{P}_{\text{net}}=(60.0,0,0)\ \mathrm{N},
$$

the steady-flow momentum equation

$$
\sum\mathbf{F}_{\mathrm{CV}}
=
\int_{\mathrm{CS}}
\rho\mathbf{V}(\mathbf{V}\cdot\mathbf{n})\,dA
$$

indicates that a net external force in the positive $x$-direction is required.

--------------------------------------

Q4. Why do we have to consider the pressure force to get the real force?

A4. 선형 운동량 방정식 $$\sum \vec{F}_\text{c.v} = \int_{\text{CS}} \vec{V} \rho(\mathbf{V} \cdot \mathbf{n}) \, dA $$의 $\sum \vec{F}_\text{c.v}$은 검사체적에 작용하는 모든 외력의 합으로, 검사체적의 출입구에 수직으로 작용하는 압력력을 고려해야 한다.

In the linear momentum equation, $\sum\mathbf{F}_{\mathrm{CV}}$ is the sum of all external forces acting on the control volume. Therefore, the pressure forces acting normal to the inlet and outlet must also be considered when determining the actual wall or support force.

# Case 3: $90^\circ$ elbow with pressure forces

### Goal

Check the effect of a change in flow direction on momentum conservation.

### Assumptions and given conditions

1. The flow is steady.
2. The fluid is incompressible.
3. The pressure is gauge pressure.
4. $\, \rho = 1000kg/m^3$
5. inlet : $ x-axis, A_1 = 0.005 m^2, \mathbf{V}_1 = 4.0 m/s, p_1 = 120 kPa$
6. outlet: : $y-axis, A_2 = 0.005 m^2, \mathbf{V}_2 = 4.0 m/s, p_2 = 80 kPa $
7. Neglect the gravity and friction in this case.

### We have to get,

1. mass flow rate, $\dot{m}$
2. net momentum flux vector, $\mathbf{P}_\text{net}$
3. inlet pressure force vector, $\mathbf{F}_\text{pressure,in}$
4. outlet pressure force vector, $\mathbf{F}_\text{pressure,out}$
5. wall force on the fluid, $\mathbf{F}_\text{wall}$
6. Force exerted by the fluid on the elbow, $\mathbf{F}_\text{fluid}$

In [4]:
# case 3

rho = 1000 # [kg/m^3]

P_1 = 120000 # [Pa]
A_1 = 0.005 # [m^2]
V_1 = 4.0 # [m/s]

P_2 = 80000 # [Pa]
A_2 = 0.005 # [m^2]
V_2 = 4.0 # [m/s]

F_1 = P_1*A_1
F_2 = P_2*A_2

m_flow_rate = rho*A_1*V_1

inlets = [ {"velocity": (V_1, 0, 0), "mass_flow_rate": m_flow_rate}]
outlets = [ {"velocity": (0, V_2, 0), "mass_flow_rate": m_flow_rate}]

net = net_momentum_flux(inlets, outlets)

F_pressure_in = np.array([F_1, 0, 0], dtype=float)
F_pressure_out = np.array([0, -F_2, 0], dtype=float)

F_wall = net - F_pressure_in - F_pressure_out
F_fluid = -F_wall

print(f"mass flow rate: {m_flow_rate}, P_net: {net}, F_in: {F_pressure_in}, F_out: {F_pressure_out}, F_wall: {F_wall}, F_fluid: {F_fluid}")

mass flow rate: 20.0, P_net: [-80.  80.   0.], F_in: [600.   0.   0.], F_out: [   0. -400.    0.], F_wall: [-680.  480.    0.], F_fluid: [ 680. -480.   -0.]


### Results

$$ \dot{m}: 20.0 \, \mathrm{kg/s}$$ $$\mathbf{P}_\text{net}: [-80.  80.   0.] \, \mathrm{N}$$ 
$$\mathbf{F}_\text{in}: [600.   0.   0.] \, \mathrm{N}, \, \mathbf{F}_\text{out}: [0. -400. 0.] \, \mathrm{N}, $$
$$\mathbf{F}_\text{wall}: [-680. 480. 0.] \, \mathrm{N}, \, \mathbf{F}_\text{fluid}: [680. -480. 0.] \, \mathrm{N}$$

### Physical Interpretation
입구와 출구의 속력은 같으나, 방향이 변하기에 순 운동량 플럭스는 0이 아니다. 각각 $+x, -y$ 방향으로 입출구에 수직으로 작용하는 압력과 외력을 성분 분해하여 외력의 방향별 성분을 쉽게 구할 수 있다. 압력력만으로는 case 3에 부합하는 운동량 변화를 만들 수 없기에, 벽이 추가적인 외력을 유체에 가해주어야 한다. 마찰력과 중력을 무시하는 case 3의 경우, 벽이 유체에 가하는 힘을 알면 작용-반작용의 원리에 의해 유체가 벽에 가하는 힘을 쉽게 알 수 있다.

The inlet and outlet speeds are the same, but the net momentum flux is not zero because the direction of the velocity changes. The pressure forces acting normal to the inlet and outlet point in the positive $x$- and negative $y$-directions, respectively. By resolving the pressure and external forces into directional components, each component of the external force can be obtained more easily.

The pressure forces alone cannot produce the momentum change required in Case 3, so the wall must exert an additional external force on the fluid. Since friction and gravity are neglected in this case, once the force exerted by the wall on the fluid is known, the force exerted by the fluid on the wall can be found directly from the action-reaction principle.


## Conclusion and limitations

### Conclusion
단면적이 일정한 직선 정상 유동인 경우 동일한 입출구 속도를 가지며 순 운동량 플럭스가 0이다.
노즐의 경우 속력이 증가하면 양의 x축 방향 운동량의 변화가 발생한다. 
방향이 변화하는 경우 같은 입출구 면적 및 속도를 가지더라도 운동량의 변화가 발생한다.

For steady, straight flow through a constant-area duct, the inlet and outlet velocities are the same, so the net momentum flux is zero. In the nozzle, an increase in speed produces a change in momentum in the positive $x$-direction. When the flow direction changes, momentum also changes even if the inlet and outlet areas and speeds are the same.

### Limitations

정상유동, 균일한 속도분포, 비압축성 유체 및 중력과 마찰을 무시하는 모델을 설정하였기에 포괄적 유체 분석을 수행하기에는 아직 어려움이 따른다. 

This model assumes steady flow, a uniform velocity distribution, and an incompressible fluid. It also neglects gravity and friction in the elbow case. Therefore, it is still difficult to use this model alone for a comprehensive analysis of real fluid flows.
